src/wallet/rpcwallet.cpp

```diff
@@ -221,10 +224,25 @@ UniValue getnewaddress(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Error: Keypool ran out, please call keypoolrefill first");
     }
     CKeyID keyID = newKey.GetID();
+    CBitcoinAddress addr = CBitcoinAddress(keyID);
 
     pwallet->SetAddressBook(keyID, strAccount, "receive");
+    bool witnessify = gArgs.GetBoolArg("-defaultwitnessaddress", false);
+    if (request.params.size() >= 2) {
```

Comment:
Remove this test, `isNull()` below is enough because `request.params[]` returns null `UniValue` for out of bounds elements.